skhep/dataset/numpydataset.py

```diff
@@ -94,6 +94,23 @@ def select(self, selection = None):
             return NumpyDataset(data, provenance)
         else:
             raise ValueError("selection input must be of type 'str', 'Selection', or an Array of booleans not {0}".format(type(selection)))
+    
+    @property        
+    def nevents(self):
+        """Get the number of events in the NumpyDataset."""
+        return self.__len__()
+     
+    @property   
+    def nentries(self):
+        """Get the number of entries in the NumpyDataset."""
+        return self.__len__()
+        
+    @property        
+    def variables(self):
+      """
+      List of variable names
```

Comment:
You could be explicit and state that the method returns contents of "numpy.dtype.names" from the stored NumPy array.